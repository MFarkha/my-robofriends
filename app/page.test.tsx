import Page from "./page";
import { expect, test } from "vitest";
import { beforeAll, afterEach, afterAll } from "vitest";
import {
  render,
  screen,
  waitForElementToBeRemoved,
  cleanup,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { setupServer } from "msw/node";
import { http, HttpResponse } from "msw";

export const handlers = [
  http.get("https://jsonplaceholder.typicode.com/users", () => {
    return HttpResponse.json([
      {
        name: "John Snow",
        id: "1",
        email: "john@company.com",
      },
      {
        name: "Kathelyn Snow",
        id: "2",
        email: "kate@company.com",
      },
    ]);
  }),
];

const server = setupServer(...handlers);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterEach(cleanup);
afterAll(() => server.close());

test("expects to render the Page component", async () => {
  const { container } = render(<Page />);
  await waitForElementToBeRemoved(() => screen.getByText(/loading/i));
  expect(container).toMatchSnapshot();
});

test("expects to search a robot", async () => {
  const { container } = render(<Page />);
  await waitForElementToBeRemoved(() => screen.getByText(/loading/i));
  await userEvent.type(
    screen.getByRole("searchbox", { name: /search/i }),
    "Kathelyn{enter}",
  );
  expect(container).toMatchSnapshot();
});
