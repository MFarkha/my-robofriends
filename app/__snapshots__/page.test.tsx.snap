// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expects to render the Page component 1`] = `
<div>
  <div
    class="tc"
  >
    <h1
      class="f1"
    >
      My Robofriends
    </h1>
    <div
      class="pa2"
    >
      <input
        aria-label="search robots"
        class="pa3 ba b--green bg-lightest-blue"
        name="searchBox"
        placeholder="search robots"
        type="search"
      />
    </div>
    <div
      style="overflow-y: scroll; border: 5px solid black; height: 800px;"
    >
      <div>
        <div
          class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        >
          <img
            alt="robots"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Frobohash.org%2F1.png%3Fsize%3D200x200&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Frobohash.org%2F1.png%3Fsize%3D200x200&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Frobohash.org%2F1.png%3Fsize%3D200x200&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div>
            <h2
              class="f5"
            >
              John Snow
            </h2>
            <p>
              john@company.com
            </p>
          </div>
        </div>
        <div
          class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        >
          <img
            alt="robots"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div>
            <h2
              class="f5"
            >
              Kathelyn Snow
            </h2>
            <p>
              kate@company.com
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`expects to search a robot 1`] = `
<div>
  <div
    class="tc"
  >
    <h1
      class="f1"
    >
      My Robofriends
    </h1>
    <div
      class="pa2"
    >
      <input
        aria-label="search robots"
        class="pa3 ba b--green bg-lightest-blue"
        name="searchBox"
        placeholder="search robots"
        type="search"
      />
    </div>
    <div
      style="overflow-y: scroll; border: 5px solid black; height: 800px;"
    >
      <div>
        <div
          class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        >
          <img
            alt="robots"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div>
            <h2
              class="f5"
            >
              Kathelyn Snow
            </h2>
            <p>
              kate@company.com
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
